src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -597,9 +607,11 @@ static bool ShouldMarkPlacementsInvalidOnFailure(DistributedExecution *execution
 static void PlacementExecutionReady(TaskPlacementExecution *placementExecution);
 static TaskExecutionState TaskExecutionStateMachine(ShardCommandExecution *
 													shardCommandExecution);
+static bool DoesHaveDependedJobs(Job *mainJob);
 static void ExtractParametersForRemoteExecution(ParamListInfo paramListInfo,
 												Oid **parameterTypes,
 												const char ***parameterValues);
+static void LogSessionId(uint64 sessionId);
```

Comment:
I found this useful during debugging. We know what command is sent over what host-port but we dont know what session is used for that.

Reply:
@marcocitus What do you think of this?